core/src/main/java/io/seata/core/rpc/netty/v1/ProtocolV1Encoder.java

```diff
@@ -92,7 +92,7 @@ public void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) {
                 if (messageType != ProtocolConstants.MSGTYPE_HEARTBEAT_REQUEST
                         && messageType != ProtocolConstants.MSGTYPE_HEARTBEAT_RESPONSE) {
                     // heartbeat has no body
-                    Serializer serializer = SerializerServiceLoader.load(SerializerType.getByCode(rpcMessage.getCodec()));
+                    Serializer serializer = SerializerServiceLoader.load(SerializerType.getByCode(rpcMessage.getCodec()), ProtocolConstants.VERSION_1);
```

Comment:
除了序列化，v0的protocolencoder和decoder在哪？因为v0是没有header这些内容的
Except for serialization, where are the protocolencoder and decoder of v0? Because v0 has no headers

Reply:
here  https://github.com/apache/incubator-seata/pull/6226
![image](https://github.com/apache/incubator-seata/assets/3788653/650c074b-3b07-439c-a7db-722a7cb54ead)
CompatibleProtocolDecoder will chose ProtocolDecoderV0,and ProtocolDecoderV0 will provide v0 parameter